lib/inspec/fetcher/git.rb

```diff
@@ -126,18 +125,26 @@ def resolved_ref
                         elsif @tag
                           resolve_ref(@tag)
                         else
-                          resolve_ref("master")
+                          resolve_ref(default_ref)
                         end
     end
 
+    def default_ref
+      shellout("git remote show #{@remote_url} | grep 'HEAD branch' | cut -d ':' -f 2").stdout&.strip
```

Comment:
I'm going to ask for two changes here:

First, the Windows test is failing because `grep` is not available on Windows. You'll need to do the searching and cutting in Ruby, which is easy enough. I found this worked, but you may be able to come up with something neater:

```
shellout("git remote show #{@remote_url}").stdout&.lines.detect { |l| l.include? "HEAD branch:" }.split(":").last.strip
```

Secondly, there is a very real chance the user may not have `git` installed on their computer. We need to handle that (and other error conditions) more gracefully. Add error handling like that used in `resolve_ref` - check the exit_status and raise an exception.

Reply:
These points are insightful. Will work on the updates.